File: Rust/Medium/WrongAnswer/optimal-partition-of-string_0.rs
impl Solution {
    pub fn partition_string(s: String) -> i32 {
        
        let mut ans = 0;
        let mut last = [0; 26];
        for (i, c) in s.chars().enumerate() {
            last[(c as u8 - b'a') as usize] = i;
        }
        let mut i = 0;
        while i < s.len() {
            let mut j = i;
            let mut k = i;
            while j < s.len() {
                k = k.max(last[(s.chars().nth(j).unwrap() as u8 - b'a') as usize]);
                if k == j {
                    break;
                }
                j += 1;
            }
            ans += 1;
            i = k + 1;
        }
        ans
    }
}

File: C++/Medium/WrongAnswer/optimal-partition-of-string_0.cpp
class Solution {
public:
    int partitionString(string s) {
        
        int n = s.size();
        vector<int> last(26, -1);
        for (int i = 0; i < n; ++i) {
            last[s[i] - 'a'] = i;
        }
        int ans = 0, start = 0, end = 0;
        for (int i = 0; i < n; ++i) {
            end = max(end, last[s[i] - 'a']);
            if (i == end) {
                ++ans;
                start = end + 1;
            }
        }
        return ans;
    }
};